### DISCUSSION TITLE
CDM Integration for TraderX Trade Processor

### DISCUSSION THREAD

**Discussion by lucaborella89** [finos/traderX#184] [Ideas]
**Goal: Integrate Common Data Model (CDM) within the TraderX trade processor module to improve trade reporting and management**

Main Flow:
1. Trade Subscription: TraderX subscribes to the trade-feed pub/sub engine to receive trade data.
2. Trade Receipt: Upon receiving a new trade, the system extracts relevant details (e.g., instrument, quantity, price).
3. Pending Trade Storage: The trade is stored in the database as a pending trade.
4. CDM Transformation: The trade data is transformed into the CDM format
5. CDM Validation: The transformed trade data is validated against the CDM schema to ensure accuracy and completeness.
6. Processed Trade Notification: The trade is marked as "processed" in the database, and a notification is sent to the appropriate trade-feed.
7. Settlement Monitoring: The system monitors the settlement status of the trade.
8. Position Recalculation: Upon settlement, the system recalculates any position changes resulting from the trade.
9. Position Update: The updated position information is persisted in the database.
10. Position Broadcast: A notification of the updated position is sent to the relevant trade-feed.

**Comment by maoo** [comment]
@tomhealey-icma - could you please review the text above and update it as you see it fit? Also, if you have any code that is already in place, please link it here. Thanks!

**Comment by iansloyan** [comment]
**### CDM PRIMER**

The [Common Domain Model (CDM)](https://cdm.finos.org) is a standardized, machine-readable, and machine-executable model that represents financial products, trades in those products, and the lifecycle events of those trades.

Links to the CDM documentation (DOCS) and GitHub repository can be found on the [CDM Homepage](https://cdm.finos.org).

An array of CDM resources can be found in the [FINOS Knowledge Base](https://resources.finos.org/category/projects-sigs/common-domain-model-cdm?projects-sigs=common-domain-model-cdm&c=cG9zdDo5OTEwNzc%3D).

ATTENTION SHOULD BE DRAWN TO THE [FOLLOWING PRESENTATION FROM EARLIER THIS YEAR FOR THOSE FRESH TO CDM](https://resources.finos.org/category/projects-sigs/common-domain-model-cdm?projects-sigs=common-domain-model-cdm&c=cG9zdDo5OTE2Mjg%3D)
